packages/valory/skills/termination_abci/behaviours.py

```diff
@@ -526,28 +589,72 @@ def is_round_ended(self, round_id: str) -> Callable[[], bool]:
         return self._is_termination_majority
 
 
-class Event(Enum):
-    """Defines the (background) round events."""
+class TerminationRound(AbstractRound):
+    """Round to act as the counterpart of the behaviour responsible for terminating the agent."""
 
-    TERMINATE = "terminate"
+    round_id = "termination_round"
 
+    def check_payload(self, payload: BaseTxPayload) -> None:
+        """No logic required here."""
+        pass
 
-class TerminationRound(CollectSameUntilThresholdRound):
-    """Defines the Termination round. The termination round is running at all times concurrently with other rounds."""
+    def process_payload(self, payload: BaseTxPayload) -> None:
+        """No logic required here."""
+        pass
 
-    round_id: str = "termination_round"
-    allowed_tx_type = TerminationPayload.transaction_type
-    payload_attribute: str = "termination_data"
-    synchronized_data_class = SynchronizedData
+    def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Enum]]:
+        """No logic required here."""
+        return None
 
-    def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Event]]:
-        """Process the end of the block."""
-        if self.threshold_reached:
-            state = self.synchronized_data.update(
-                synchronized_data_class=self.synchronized_data_class,
-                termination_majority_reached=True,
-                most_voted_tx_hash=self.most_voted_payload,
-            )
-            return state, Event.TERMINATE
 
-        return None
+class TerminationBehaviour(BaseBehaviour):
+    """Behaviour responsible for terminating the agent."""
+
+    state_id: str = "termination"
+    behaviour_id = "termination_behaviour"
+    matching_round = TerminationRound
+
+    def async_act(self) -> Generator:
+        """Logs termination and terminates."""
+        self.context.logger.info("Terminating the agent.")
+        ok = 0
+        quit(ok)
+        yield
```

Comment:
what does `quit()` do?

For now it's ok probably, but we need to do a bit more cleanup. Can do later.

Reply:
Same as `exit()`, it will raise a `SystemExit` exception. I am open to alternatives.

Reply:
For now ok - but open an issue as I want to extend this later.